### PR TITLE
Improve arguments parsing and add -h option

### PR DESCRIPTION
**Pull request by StefanosChaliasos**

### CODE REVIEW

File: src/csmake.pl [hunk 1]
```diff
@@ -54,33 +54,39 @@
...
-            undef $MAKEARGS[$i];
-            undef $MAKEARGS[$i+1];
+my $csmake_opts = "hAN:T:";
+getopts($csmake_opts, \%options);
+
+my ($index) = grep { $TEMP_ARGV[$_] eq '--' } (0 .. @TEMP_ARGV-1);
+my @MAKEARGS;
+
+# Parse command line arguments and separate them into csmake arguments and make arguments.
+if ($index > 0) {
```
Comment (dspinellis) [inline src/csmake.pl]: This is a simple array splicing operation.

File: src/csmake.pl [hunk 2]
```diff
@@ -54,33 +54,39 @@
...
+}
+getopts($csmake_opts, \%options);
+
+if (defined $options{h}) {
+print <<HELP;
+usage: csmake [ [-A] [-T spy_directory] [-N rules_file] [-h] -- ] [make(1) options]
+    -A                      Generate cs projects for static libraries.
+    -N rules_file           Run on an existing rules file.
+    -T spy_directory        Create a separate CScout .cs file for each real executable.
+HELP
```
Comment (dspinellis) [inline src/csmake.pl]: `-h print this message`

I also propose to introduce two switches in place of **-A**:
* **-x** Create projects for executable files (those without an extension)
* **-o** *e* Create projects for other files with the specified extension e (can be specified multiple times, e.g. `-o so` `-o a`)

Also, a **-s** option (in place of **-T**) should specify the separate `.cs` per project directory, not the temporary directory.  A separate option, **-d** can be used to specify to leave the temporary directory intact for debugging purposes. Finally, a **-r** *path* option can be used to specify where to save the rules file.

Comment (StefanosChaliasos) [reply src/csmake.pl]: I think this [commit](https://github.com/dspinellis/cscout/pull/58/commits/0a2881f516f89668b9e5ccdc67f2457e6d3e5080) address your comments.

File: src/csmake.pl [hunk 3]
```diff
@@ -79,12 +79,20 @@
...
-usage: csmake [ [-A] [-T spy_directory] [-N rules_file] [-h] -- ] [make(1) options]
+usage: csmake [ [-A] [-d] [-k] [-s cs_files_directory] [-T spy_directory] [-N rules_file] [-h] -- ] [make(1) options]
+    -d                      Run debug mode (it also keeps spy directory in place).
+    -h                      Prints help message.
+    -k                      Keep temporary directory in place.
+    -s cs_files_directory   Create a separate CScout .cs file for each real executable.
     -A                      Generate cs projects for static libraries.
     -N rules_file           Run on an existing rules file.
-    -T spy_directory        Create a separate CScout .cs file for each real executable.
+    -T spy_directory        Set spy directory.
```
[the remark this answers is not in the document]
Comment (StefanosChaliasos) [reply src/csmake.pl]: Done.

**Comment by dspinellis** [comment]
Thank you!

**Pull request merged**